### DISCUSSION TITLE
Unable to connect VS Code using IBMi extension

### DISCUSSION THREAD

**Discussion by kmepgm** [codefori/vscode-ibmi#1716] [Q&A]
Downloaded VS Code and I am able to connect with ACS using client VPN.  Others in group can use RDI, but VS Code not connecting.  Any tips on settings to check or debugs to use.

TIA

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.5.0|
|Visual Studio Code version|1.84.2|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Emmet (emmet): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
JSON Language Features (json-language-features): 1.0.0
Material Icon Theme (material-icon-theme): 4.32.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Prettier - Code formatter (prettier-vscode): 10.1.0
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>

<hr />

**_Not connected_** 🔌

**Comment by worksofliam** [comment]
Are you able to connect using the `ssh` command from a terminal?

**Comment by sebjulliand** [comment]
Hi!
ACS emulator connects through telnet, RDi through JT400's connections, and Code for i through SSH. So it's most likely that your SSH configuration is at fault here. Please check out the documentation about getting started and tips about how to correctly configure your IBM I. 
You can also check out IBM i OSS documentation, especially the getting started section (https://ibmi-oss-docs.readthedocs.io/en/latest/yum/README.html) as well as the Setting up your environment section (https://ibmi-oss-docs.readthedocs.io/en/latest/user_setup/README.html)

**Reply by NextPage66** [reply]
I also have the same connection issue.  I was connected and using VS Code occasionally about 4 weeks ago.  I have since upgraded IBM i to 7.4 and now VS Code will not connect.  I have SSH running just like it was on 7.3.  I deleted and reloaded VS Code and Code for IBM i.  No luck.  I will read through the docs but any ideas to help would be appreciated.

**Reply by NextPage66** [reply]
Message says "Error running command code-for-ibmi.connect: command 'code-for-ibmi.connect' not found. This is likely caused by the extension that contributes code-for-ibmi.connect."

**Reply by worksofliam** [reply]
@NextPage66 can you confirm with us what version of Code for IBM i version you're using? 2.6.1 was just released.

**Reply by NextPage66** [reply]
I have installed several versions trying to get it to work.  I went back to 1.6.15 and it would not work.  I currently have 2.6.1 installed.

**Reply by NextPage66** [reply]
I have installed PuTTy and connected a bash terminal.  So I think the SSH port is open and working on the IBM i.

**Reply by sebjulliand** [reply]
@NextPage66 could you open the dev tools console and paste it's content after you see that "command not found" message?
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/90cb402a-7ba8-44dd-bc99-682853156f32)
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/73680931-d469-4b77-b897-bac3e9b079e6)

Thanks!

**Reply by worksofliam** [reply]
Sounds to me like you might be in an untrusted workspace, which I believe stops some commands from working correctly.

https://code.visualstudio.com/docs/editor/workspace-trust

**Reply by NextPage66** [reply]
WARN Via 'product.json#extensionEnabledApiProposals' extension 'ms-python.python' wants API proposal 'registerIssueDataProvider' but that proposal DOES NOT EXIST. Likely, the proposal has been finalized (check 'vscode.d.ts') or was abandoned.
log.ts:421  INFO Started local extension host with pid 19636.
console.ts:137 [Extension Host] Git installation not found.
w @ console.ts:137
mainThreadExtensionService.ts:113 Activating extension 'halcyontechltd.vscode-ibmi-walkthroughs' failed: Cannot find module 'c:\Users\dsanders\.vscode\extensions\halcyontechltd.vscode-ibmi-walkthroughs-0.4.0\extension.js'
Require stack:
- c:\Users\dsanders\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\loader.js
- c:\Users\dsanders\AppData\Local\Programs\Microsoft VS Code\resources\app\out\bootstrap-amd.js
- c:\Users\dsanders\AppData\Local\Programs\Microsoft VS Code\resources\app\out\bootstrap-fork.js
- .
$onExtensionActivationError @ mainThreadExtensionService.ts:113
mainThreadExtensionService.ts:113 Cannot activate the 'Code for IBM i' extension because its dependency 'Code for IBM i Walkthroughs' failed to activate
$onExtensionActivationError @ mainThreadExtensionService.ts:113
log.ts:421  INFO [perf] Render performance baseline is 40ms
webviewElement.ts:513 An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can escape its sandboxing.
mountTo @ webviewElement.ts:513
webviewElement.ts:513 An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can escape its sandboxing.
mountTo @ webviewElement.ts:513
icon.png:1     Failed to load resource: the server responded with a status of 404 ()
log.ts:441   ERR command 'code-for-ibmi.connect' not found: Error: command 'code-for-ibmi.connect' not found
    at k.n (vscode-file://vscode-app/c:/Users/dsanders/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1538:12339)
    at k.executeCommand (vscode-file://vscode-app/c:/Users/dsanders/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1538:12268)
    at async a.open (vscode-file://vscode-app/c:/Users/dsanders/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:645:43176)
    at async f.open (vscode-file://vscode-app/c:/Users/dsanders/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:645:44557)
notificationsAlerts.ts:42 Error running command code-for-ibmi.connect: command 'code-for-ibmi.connect' not found. This is likely caused by the extension that contributes code-for-ibmi.connect.
c @ notificationsAlerts.ts:42

**Reply by worksofliam** [reply]
@NextPage66 From your extensions view, do you have the Code for IBM i Walkthroughs extension installed?

**Reply by NextPage66** [reply]
Yes.  I didn't know the log existed.  I am very new to this platform.  I downgraded to previous version of Code for IBM i Walkthroughs and the connection is working now.

**Reply by NextPage66** [reply]
If i have a question on a closed issue for Code for IBM i, what do I do?
